features/vesting/hooks.ts

```diff
@@ -173,3 +243,45 @@ export const useVestingSnapshotDelegate = () => {
     [contractWeb3, chainId],
   );
 };
+
+export const useRevokeUnvested = (escrow: string | undefined) => {
+  const { chainId } = useWeb3();
+  const { contractWeb3 } = useVestingEscrowContract(escrow);
+
+  return useCallback(async () => {
+    if (chainId == null || contractWeb3 == null) {
+      return;
+    }
+    await transaction('Revoke unvested tokens', chainId, () =>
+      contractWeb3['revoke_unvested'](),
+    );
+  }, [chainId, contractWeb3]);
+};
+
+export const useVestingIsRevoked = (escrow: string | undefined) => {
+  const { chainId } = useWeb3();
+  const { contractRpc } = useVestingEscrowContract(escrow);
+
+  const getEvents = useCallback(
+    async (chainId?: CHAINS) => {
+      if (chainId == null) return false;
+
+      const events = await contractRpc.queryFilter(
+        contractRpc.filters.UnvestedTokensRevoked(),
+        EVENTS_STARTING_BLOCK[chainId],
+      );
+
+      return events.length > 0;
```

Comment:
this FN should be probably named `getHasSomeEvents`

Reply:
Makes sense